.travis.yml

```diff
@@ -1,6 +1,21 @@
 services:
   - docker
 
-script:
-  - docker build --target test --tag georgefrancis:test0.1 .
-  - docker run --mount type=bind,source="$(pwd)",target=/app --env TRELLO_KEY --env TRELLO_TKN --env TRELLO_BOARD georgefrancis:test0.1 .
\ No newline at end of file
+jobs:
+  include:
+    - stage: test
+      script:
+        - docker build --target test --tag devops-starter:test0.1 .
+        - docker run --mount type=bind,source="$(pwd)",target=/app --env TRELLO_KEY --env TRELLO_TKN --env TRELLO_BOARD devops-starter:test0.1 .
+    - stage: build-tag-push
+      if: branch = master AND type = push
```

Comment:
Nice work spotting that this needs to be master _and_ a push rather than pull - an alternative solution is to take advantage of the `deploy` lifecycle stage from Travis, which won't execute on PRs, and for which we can apply a similar branch condition; there's no big advantage other than semantically making it clear what the intention is